library/kotlin/io/envoyproxy/envoymobile/RequestBuilder.kt

```diff
@@ -0,0 +1,90 @@
+package io.envoyproxy.envoymobile
+
+import java.net.URL
+
+
+/**
+ * Builder used for constructing instances of `Request` types.
+ *
+ * @param url URL for the request.
+ * @param method Method for the request.
+ */
+class RequestBuilder(
+    val url: URL,
+    val method: RequestMethod
+) {
+  // Headers to send with the request.
+  // Multiple values for a given name are valid, and will be sent as comma-separated values.
+  private val headers: MutableMap<String, MutableList<String>> = mutableMapOf()
+
+  // Trailers to send with the request.
+  // Multiple values for a given name are valid, and will be sent as comma-separated values.
+  private val trailers: MutableMap<String, MutableList<String>> = mutableMapOf()
+
+  // Serialized data to send as the body of the request.
+  private var body: ByteArray? = null
+
+  // Retry policy to use for this request.
+  private var retryPolicy: RetryPolicy? = null
+
+  fun addBody(body: ByteArray?): RequestBuilder {
+    this.body = body
+    return this
+  }
+
+  fun addRetryPolicy(retryPolicy: RetryPolicy?): RequestBuilder {
+    this.retryPolicy = retryPolicy
+    return this
+  }
+
+  fun addHeader(header: String, value: String): RequestBuilder {
```

Comment:
I think the general semantic here is `name` instead of `header` (i.e., the key that goes with the value). Same for trailers